### DISCUSSION TITLE
Merge queue alerting, notifications, and integration with third party incident management software

### DISCUSSION THREAD

**Discussion by amccabe-astronomer** [community/community#49523] [Pull Requests]
### Select Topic Area

Product Feedback

### Body

I have been a relatively happy consumers of Merge Queues, but there is a major feature gap that is really hurting me and my employer.

Since merge queues are a critical pieces of our CI/CD architecture, we have an on-call rotation supporting it, using typical incident management and monitoring tools.

It would be extremely useful if there were settings to configure webhooks for certain bad merge queue states, like

* When more than a certain threshold of PRs have been removed (timeout and/or failure) from the merge queue in a period of time
* If a certain number of PRs  have failed to merge consecutively

It would be better if I could ingest logs or metrics directly and implement my own alerting, but right now I am stuck with managing systems downstream of merge queues to infer issues. This is a real problem for managing an internal SLA for pipeline performances.

**Comment by ghost** [comment]
+1 to this. I'm looking for ways to monitor/measure merge queue performance. I'd like to answer questions like:
1. What is the median wait time to be merged?
2. What is the pass/fail rate for merge attempts?
3. What is the median merge time itself? (That is, once the merge test has begun, how long until the code is actually merged?)

For all these questions, I want time series data so I can analyze merge queue behavior over time. I've searched around, and thus not found that this underlying data is exported in a discoverable location.

**Reply by jaimebarriga** [reply]
The approach we have taken was to send the webhooks from Github relating to the merge queue (`pull_request.enqueue`, `pull_requests.dequeue`, etc...) to a separate service that then uses that to calculate certain stats that are important to us and sending them to Datadog.

Less than ideal though, wish some of these stats were calculated and tracked in Github.
